File: app/api/stats/stats.js
'use strict';

async function getStats() {
    try {
        // Use dynamic import with await
        const { default: fetch } = await import('node-fetch');

        const API_KEY_SECRET = 'hispacep2p_default_secret';
        const hispace_URL = 'https://p2p.hispace.com/api/v1/stats';
        //const hispace_URL = 'http://localhost:3000/api/v1/stats';

        const response = await fetch(hispace_URL, {
            method: 'GET',
            headers: {
                authorization: API_KEY_SECRET,
                'Content-Type': 'application/json',
            },
        });
        const data = await response.json();
        if (data.error) {
            console.log('Error:', data.error);
        } else {
            if (data) {
                const formattedData = JSON.stringify(data, null, 2);
                console.log(formattedData);
            }
        }
    } catch (error) {
        console.error('Error fetching data:', error);
    }
}

getStats();

File: app/api/stats/stats.py
# pip3 install requests
import requests
import json

API_KEY_SECRET = "hispacep2p_default_secret"
hispace_URL = "https://p2p.hispace.com/api/v1/stats"
#hispace_URL = "http://localhost:3000/api/v1/stats"

headers = {
    "authorization": API_KEY_SECRET,
    "Content-Type": "application/json",
}

response = requests.get(
    hispace_URL,
    headers=headers
)

print("Status code:", response.status_code)

if response.status_code == 200:
    data = response.json()
    pretty_printed_data = json.dumps(data, indent=4)
    print(data)
else:
    print("Failed to retrieve data. Error:", response.text)
